services/motion/builtin/builtin.go

```diff
@@ -385,30 +483,43 @@ func (ms *builtIn) planMoveOnGlobe(
 		return nil, nil, err
 	}
 
-	// make call to motionplan
-	solutionMap, err := motionplan.PlanMotion(
-		ctx,
-		ms.logger,
-		referenceframe.NewPoseInFrame(referenceframe.World, goal),
-		offsetFrame,
-		inputMap,
-		fs,
-		wrldst,
-		nil,
-		extra,
-	)
-	if err != nil {
-		return nil, nil, err
-	}
+	return &motionplan.PlanRequest{
+		Logger:             ms.logger,
+		Goal:               referenceframe.NewPoseInFrame(referenceframe.World, goal),
+		Frame:              offsetFrame,
+		FrameSystem:        fs,
+		StartConfiguration: referenceframe.StartPositions(fs),
+		WorldState:         worldState,
+		Options:            extra,
+	}, kb, nil
+}
 
-	plan, err := motionplan.FrameStepsFromRobotPath(kbf.Name(), solutionMap)
+func (ms *builtIn) executePlan(ctx context.Context, kinematicBase kinematicbase.KinematicBase, plan motionplan.Plan) error {
+	waypoints, err := plan.GetFrameSteps(kinematicBase.Name().Name)
 	if err != nil {
-		return nil, nil, err
+		return err
 	}
-	for _, step := range plan {
-		ms.logger.Info(step)
+
+	for i := 1; i < len(waypoints); i++ {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			ms.logger.Info(waypoints[i])
+			segment := []referenceframe.Input{
+				{Value: waypoints[i][0].Value - waypoints[i-1][0].Value},
+				{Value: waypoints[i][1].Value - waypoints[i-1][1].Value},
+			}
+			if err := kinematicBase.GoToInputs(ctx, segment); err != nil {
+				// If there is an error on GoToInputs, stop the component if possible before returning the error
+				if stopErr := kinematicBase.Stop(ctx, nil); stopErr != nil {
```

Comment:
[q] I do not understand how this would not fail should be choose to cancel the context due to a replan event